src/main/java/jenkins/plugins/openstack/compute/SectionDisabled.java

```diff
@@ -0,0 +1,277 @@
+package jenkins.plugins.openstack.compute;
+
+import java.io.Serializable;
+import java.util.concurrent.TimeUnit;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+import org.kohsuke.stapler.QueryParameter;
+
+import hudson.Extension;
+import hudson.Functions;
+import hudson.Util;
+import hudson.model.AbstractDescribableImpl;
+import hudson.model.Descriptor;
+import hudson.util.FormValidation;
+
+/**
+ * Records that the user has disabled something "until further notice", or the
+ * system has disabled something for a period, or both.
+ */
+public class SectionDisabled extends AbstractDescribableImpl<SectionDisabled> implements Serializable {
+
+    private static final long serialVersionUID = 1L;
+
+    /** Value returned by {@link #toString()} if {@link #isDisabled()} is false. */
+    @Restricted(NoExternalUse.class)
+    static final String NOT_DISABLED = "No";
```

Comment:
I would prefer this would be a flyweight instance of `SectionDisabled` so we do not have to do the string-based comparison.

Reply:
Yes, I'd agree that this is a bit messy ... I don't have the time/resources to dive into the specifics right now but from what I can remember, if there's a neater method then that'd be fine by me.